### PR TITLE
Reuse common configurations

### PR DESCRIPTION
**Pull request by Goooler**
## Checklist

- [ ] PR description added
- [ ] tests are added
- [ ] KtLint has been applied on source code itself and violations are fixed
- [ ] [documentation](https://pinterest.github.io/ktlint/) is updated
- [ ] `CHANGELOG.md` is updated

In case of adding a new rule:
- [ ] Rule is added to [rules documentation](https://pinterest.github.io/ktlint/rules/standard/)

### CODE REVIEW

**Comment by Goooler** [comment]
Depends on #1891.

File: .github/workflows/gradle-pr-build.yml [hunk 1]
```diff
@@ -35,10 +35,3 @@ jobs:
         run: ./gradlew build ktlintCheck
       - name: Build with dev Kotlin version
         run: ./gradlew -PkotlinDev build ktlintCheck
-      - name: Upload test results
-        uses: actions/upload-artifact@v3
-        if: failure()
-        with:
-          name: Test artifacts
-          path: '**/build/reports/tests/**'
-          if-no-files-found: warn
```
Comment (Goooler) [inline .github/workflows/gradle-pr-build.yml:44]: I'm removing this action, you can view test results via Gradle build scan.

Comment (Goooler) [inline .github/workflows/gradle-pr-build.yml:44]: I'm removing this action, you can view test results via Gradle build scan.

Comment (Goooler) [inline .github/workflows/gradle-pr-build.yml:44]: I'm removing this action, you can view test results via Gradle build scan for now.

File: ktlint-api-consumer/build.gradle.kts [hunk 2]
```diff
@@ -5,7 +5,7 @@ plugins {
 dependencies {
     // Any SLF4J compatible logging framework can be used. The "slf4j-simple" logging provider is configured in file
     // ktlint-api-consumer/src/main/resources/simplelogger.properties
-    runtimeOnly("org.slf4j:slf4j-simple:2.0.7")
+    runtimeOnly(libs.slf4j)
```
Comment (paul-dingemans) [inline ktlint-api-consumer/build.gradle.kts:8]: Please revert this change. I intended this module to be an easy example of how an API consumer can look like. So the slf4j dependency is not meant to be used anywhere else in the project.

Maybe something similar should also be done for line 10 and 11 as well as for other modules. We might need to distinct between modules meant for publication as artifact (e.g. being a part of the Ktlint API or CLI) and other modules that are only meant for clarification. The latter group of modules should not reuse any of the common build logic but be a standalone build script. What do you think?

Comment (Goooler) [reply ktlint-api-consumer/build.gradle.kts]: cb9b94a0c80157fa858c89ece8e8357ecac50bd4

File: .github/workflows/gradle-pr-build.yml (hunk 1 above)
Comment (paul-dingemans) [reply .github/workflows/gradle-pr-build.yml:44]: I am not liking this. I don't like to fill in my email adress each time that I want to get access to a build scan. I am developing using a laptop from my employer. I don't want to use my work email adress to get access to the scan. Neither do I want to access my personal email on my work laptop.

What would be the benefit of removing it?

Comment (Goooler) [reply .github/workflows/gradle-pr-build.yml:44]: It's easier to view and share build scans, I noticed I have no need to fill email address to view build scans that had been published on CI, you can have a try like https://gradle.com/s/tcalcngp6a3tk.

Comment (paul-dingemans) [reply .github/workflows/gradle-pr-build.yml:44]: Yes, I just noticed that too in another failed build. And in no way, I meant that the Gradle build scan had to be removed ;-)
So let try it out then.

File: build.gradle.kts [hunk 3]
```diff
@@ -6,27 +6,6 @@ plugins {
     alias(libs.plugins.shadow) apply false
 }
 
-allprojects {
-    if (project.hasProperty("isKotlinDev")) {
-        val definedVersion = ext["VERSION_NAME"].toString().removeSuffix("-SNAPSHOT")
-        ext["VERSION_NAME"] = "$definedVersion-kotlin-dev-SNAPSHOT"
-    }
```
Comment (Goooler) [inline build.gradle.kts:13]: Killing `allprojects`.

File: build-logic/src/main/kotlin/ktlint-publication.gradle.kts [hunk 4]
```diff
@@ -5,6 +5,11 @@ plugins {
     signing
 }
 
+if (project.hasProperty("isKotlinDev")) {
+    val definedVersion = ext["VERSION_NAME"].toString().removeSuffix("-SNAPSHOT")
+    ext["VERSION_NAME"] = "$definedVersion-kotlin-dev-SNAPSHOT"
+}
```
Comment (Goooler) [inline build-logic/src/main/kotlin/ktlint-publication.gradle.kts:11]: All modules need to be published depending on `ktlint-publication.gradle.kts`.

File: ktlint-api-consumer/build.gradle.kts (hunk 2 above)
Comment (Goooler) [reply ktlint-api-consumer/build.gradle.kts]: I still we should put this dep into version catalog, we should not reuse publication logic but deps.

**Pull request merged**